contracts/multisig/src/key.rs

```diff
@@ -13,8 +14,43 @@ pub enum KeyType {
 
 #[cw_serde]
 #[derive(PartialOrd, Ord, Eq)]
-pub enum Signature {
-    Ecdsa(HexBinary),
+pub enum Signature<T>
+where
+    T: Recoverability,
```

Comment:
This is adding more complexity to the Signature type. Recoverability is not core concept for a Signature scheme. It's very EVM specific. ed25519 doesn't have a recoverable variant for e.g.

Since recoverable signatures are only needed when encoding the proof, I'd suggest just using plain bytes for the type